### DISCUSSION TITLE
役に立つ情報を共有しよう

### DISCUSSION THREAD

**Discussion by tomishima2904** [tomishima2904/shapley_association_game#14] [General]
参考文献や拡張機能等開発に役立ったものはここに何でも投下しよう

**Comment by tomishima2904** [comment]
Vscodeの拡張機能のLive serverってやつがフロントエンド開発だとかなり便利

**Comment by tomishima2904** [comment]
[Djangoで既存のDBを使う方法](https://qiita.com/bow_arrow/items/2641267e1e088ef58d18)
```
python manage.py inspectdb >> models.pyへのパス/models.py
```

**Comment by tomishima2904** [comment]
スーパーユーザーの追加方法
```
python manage.py createsuperuser 
```

**Reply by tomishima2904** [reply]
スーパーユーザーを追加した後、ブラウザのアドレスバーに
```
http://localhost:8000/admin
```
を入力してアクセスするとデータベースの管理者画面に移れる。
<img width="573" alt="スクリーンショット 2022-12-09 11 15 04" src="https://user-images.githubusercontent.com/49580272/206608628-5a79eabf-f794-4ff7-b710-f3898be83f2a.png">

**Comment by tomishima2904** [comment]
【認証系のフロントエンドに関して】
認証系のテンプレートを改変するときは、`djnago-allauth`で提供されているテンプレート名に遵守し、[提供元](https://github.com/pennersr/django-allauth/tree/master/allauth/templates/account)のソースコードを参考にすると良い。

**Reply by tomishima2904** [reply]
【認証系テンプレートの改変方法 #2 】
デフォルトでは[ここ](https://github.com/pennersr/django-allauth/tree/master/allauth/templates/account)のテンプレートが適用されているので、もし変えたい場合は以下のようにする
- `accounts/templates`に`account`フォルダを作成
- `account`フォルダ内に改変したいテンプレートを作成する。この際、ログイン画面ならば`login.html`、メールの認証画面ならば`email_confirm.html`のように[提供元](https://github.com/pennersr/django-allauth/tree/master/allauth/templates/account)の**名前と同じでなければならない**。
(例えば、ログイン画面を`login2.html`とか`hoge.html`で作成すると反映されない。正しい名前で。)
- htmlを作成したら勝手に作成した方のhtmlがアプリの方に反映されるので自由に改変しよう。**ただし、提供元に備わっている機能は最低限つけておいてほしい**。

こちらでも、背景のデザインとかフォントとか、どのページでも共通するようなものは`base.html`で定義しちゃうと楽かな

**Comment by tomishima2904** [comment]
【認証系に関して】
`http://localhost:8000/accounts/login/`にアクセスすると`django-allauth`が提供しているログイン画面にアクセスできる。

<img width="620" alt="スクリーンショット 2022-12-09 11 16 43" src="https://user-images.githubusercontent.com/49580272/206608865-7b699836-41d6-4a99-a51d-aa68ac875dcb.png">

↑添付の画像のようにデフォルトは質素なのでできるればデザインを変更したい。
変更方法に関しては[こちら](https://github.com/tomishima2904/shapley_association_game/discussions/14#discussioncomment-4328631)のディスカッションを参照

**Reply by tomishima2904** [reply]
【ユーザーの追加方法】
1. `docker-compose up web`でwebのコンテナを立ち上げる. `-d`をつけないで立ち上げるように！
2. `http://localhost:8000/accounts/login/`にアクセスして、`ユーザー登録`リンクを押す
3. メールアドレス、ユーザー名、パスワードを入力して登録する
4. ターミナル上にメールアドレス認証用のURLが送信されるのでそれをブラウザにコピペする
5. 確認ボタンを押せばユーザーが完了するので、ログイン画面で先ほど登録したメアドとパスワードを打ち込んでログインしよう

**Reply by yuma283** [reply]
メールアドレスはとりあえず架空のものでも、既存のものでも構わない（テストする時など）
本番では実際に使われているメアドを登録してもらう

**Comment by tomishima2904** [comment]
Tailwind CSSを使うと見出しタグ`<h1>`とか、`<a>`タグの通常のスタイルが適用されないので注意

**Comment by tomishima2904** [comment]
【デザインに関して】
[より良いデザインにするためのHTML・CSS](https://ics.media/entry/221208/)

**Comment by ttoitt** [comment]
必要かわからないけど、以前にDiscordで議論した実験材料についてのやり取りをここにも載せておく

Tomishima Ryo — 2022/11/22 19:37
@toi takumi 
インフラ周りを一緒にやってもらおうと思ってるけど、インフラの仕事はまだまだ先になりそう. なので
・どんな感じでユーザーに答えて欲しいか
・どういう実験材料が欲しいのか
とかを教えてくれると助かる

toi takumi — 2022/11/22 20:34
・どんな感じでユーザーに答えて欲しいか
正解語を予測したときに、人間と機械で根拠とした語がどう違うかを見たいと思ってる。
だから、選択肢提示してその中から選んでもらうようにしたい。
こんな感じかな
添付ファイル種類：acrobat
B1___200914.pdf
342.49 KB

Tomishima Ryo — 2022/11/22 20:38
選択肢というのはユーザーが解答するのに役に立った刺激語を選択できるようにってことだよね？

toi takumi — 2022/11/22 20:38
あ、ごめん
選択肢は正解語の方
正解語答えてもらったあと、答えるのに役立った刺激語を順に並べてもらうことを考えてる。

Tomishima Ryo — 2022/11/22 20:45
なるほどー
じゃあ機能的に役に立った刺激語の順番もユーザーには答えてもらおうか

toi takumi — 2022/11/22 20:48
そうなる
実験材料は喚語資料を使う予定

Tomishima Ryo — 2022/11/22 20:51
喚語資料の中の問題全部使う？もしFITで使ったデータセットだけとかで良いならそのテキストファイルとかくれると助かる！

toi takumi — 2022/11/22 20:58
これ！
計87問
添付ファイル種類：spreadsheet
keywords_fit2022.csv
5.46 KB

Tomishima Ryo — 2022/11/22 20:58
おけい！サンクス！

**Reply by ttoitt** [reply]
ユーザに応えてもらう画面の概要のファイル（喚語資料：B1' まとめる想起（同一カテゴリー）選択肢あり）
実験材料（僕がFIT2022で発表したときに使った87題、黒岩研24卒の3人が解いて答えがある程度一致したもの）

ファイルが載せられない？ので文章で説明

**Reply by tomishima2904** [reply]
Discord上のリンク集に、共有してあるGoogle Driveがあるので、そこの`data`ディレクトリに入れてもらえると助かる！

**Reply by ttoitt** [reply]
了解！

**Comment by yuma283** [comment]
tailwind CSSで使えそうなサイトをいくつか貼っておく。
一番下のリンクはチートシートなので、何かクラスを調べたい時などに使えるかも
https://reffect.co.jp/html/tailwindcss-for-beginners
https://codelikes.com/tailwindcss-used-class/
https://flowbite.com/tools/tailwind-cheat-sheet/

**Comment by yuma283** [comment]
公式のチートシートがあったので共有(https://tailwindcss.com/docs/align-items#center)
こちらは簡易的な日本語チートシート、基礎的なことが書いてある(https://uedive.net/2021/5596/tailwind-css/#st-toc-h-12)
こちらはtailwindCSSでよく使うデザインパターンだそう、もし詰まったら参考に(https://jsnotice.com/posts/2020-04-30/)
（ボタンの上下中央揃えですら、知らないと難しかった）

**Reply by tomishima2904** [reply]
- [Tailwind CSSの一歩進んだ書き方](https://zenn.dev/ixkaito/articles/advanced-tailwindcss)

**Reply by tomishima2904** [reply]
`@apply`を使うとよく使う表現(ボタンとか)を使いまわせるようになる。
- [公式](https://tailwindcss.com/docs/functions-and-directives#apply)
- [【TailwindCSS】@applyと@layerを使用してカスタムCSSクラスを作る](https://itosae.com/tailwindcss_lesson_1/)

**Reply by tomishima2904** [reply]
<img width="769" alt="スクリーンショット 2022-12-16 15 49 49" src="https://user-images.githubusercontent.com/49580272/208039569-90af531b-34ef-4822-bc11-c43bc37018a5.png">

**Reply by tomishima2904** [reply]
【重要】
テンプレート（html）をいじる際は必ず下記のコードを実行してからやろう！（コンテナに入らなくて平気）
```
npm run dev
```
HTMLファイルの変更をリアルタイムでウォッチしてそれを`style_dist.css`に反映させている。

**Reply by tomishima2904** [reply]
[いい感じのコンポーネント集](https://tailwind-elements.com/)

**Reply by tomishima2904** [reply]
`npm run dev`をやると以下のようになる
```
shapley_association_game % npm run dev

> dev
> npx tailwindcss -i static/css/style.css -o static/css/style_dist.css --watch


Rebuilding...

Done in 2800ms.
```

**Reply by tomishima2904** [reply]
[フリー素材のアイコン集](https://fonts.google.com/icons)

**Reply by tomishima2904** [reply]
【tailwindcssを使う上で役に立つvscodeの拡張機能】
名前: Tailwind CSS IntelliSense
ID: bradlc.vscode-tailwindcss

tailwindcssの予測変換してくれる

**Reply by tomishima2904** [reply]
- [良さげなGoogle fonts集](https://photoshopvip.net/131206)
- [こちら](https://fonts.google.com/?query=kosu&preview.text=%E4%BB%B2%E9%96%93%E3%81%AF%E3%81%AA%E3%82%93%E3%81%A7%E3%81%99%E3%81%8B%EF%BC%9F&preview.text_type=custom)でプレビュー見ることができる

**Comment by tomishima2904** [comment]
【Django × フロントエンド】
HTMLを書くとき、Djangoでは[Djangoテンプレート言語](https://docs.djangoproject.com/ja/3.2/ref/templates/builtins/)というものが提供されている。これを使えば、`if`や`for`も使うことができる。

**Comment by tomishima2904** [comment]
【Django → Javascript】
Djangoのviewsからのレスポンス（json型データ）をJavascriptで受け取るやり方は[こちら](https://teratail.com/questions/299286)

**Comment by tomishima2904** [comment]
【DBのバックアップ】
```
python manage.py backup_gameresults
```

**Comment by tomishima2904** [comment]
【質問用データベースの更新手順】
1. DB側のコンテナを立ち上げる `docker-compose up -d db`
2. DBのコンテナにSQLファイルを転送する `docker cp data/words.sql $(docker-compose ps -q db):/tmp/words.sql`
3. DB側のコンテナに入る `docker-compose exec db bash`
4. MySQLにSQLファイルの情報をインプットする `mysql < tmp/words.sql`

これで大丈夫な気がするが、ダメなら`readme`の
### Djangoのコンテナ側での作業
ってやつをやってみてくれい

**Comment by tomishima2904** [comment]
【gitでバグったら】
```
git fsck
```
